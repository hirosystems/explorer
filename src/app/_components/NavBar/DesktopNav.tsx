import { FC } from 'react';
import { PiCaretDown } from 'react-icons/pi';

import { Flex } from '../../../ui/Flex';
import { Icon } from '../../../ui/Icon';
import { Link } from '../../../ui/Link';
import { Popover } from '../../../ui/Popover';
import { PopoverContent } from '../../../ui/PopoverContent';
import { PopoverTrigger } from '../../../ui/PopoverTrigger';
import { Text } from '../../../ui/Text';
import { DesktopSubNav } from './DesktopSubNav';
import { NavItem } from './types';

export const DesktopNav: FC<{ navItems: NavItem[] }> = ({ navItems }) => {
  return (
    <Flex gap={6}>
      {navItems.map(navItem => (
        <Flex key={navItem.id} alignItems={'center'}>
          <Popover trigger={'hover'} placement={'bottom-start'} isLazy>
            <PopoverTrigger>
              <Flex
                gap={1.5}
                border={
                  navItem.id === 'network'
                    ? '1px solid var(--stacks-colors-whiteAlpha-600)'
                    : undefined
                }
                bg={navItem.id === 'network' ? 'whiteAlpha.200' : undefined}
                px={navItem.id === 'network' ? 3 : undefined}
                py={navItem.id === 'network' ? 1.5 : undefined}
                rounded={navItem.id === 'network' ? 'full' : undefined}
              >
                <Link
                  href={navItem.href ?? '#'}
                  color={'slate.50'}
                  fontSize="xs"
                  fontWeight={'medium'}
                  _hover={{
                    textDecoration: navItem.children ? 'none' : 'underline',
                  }}
                >
                  {navItem.label}
                </Link>
                {navItem.children && <Icon as={PiCaretDown} size={3.5} color={'white'} />}
              </Flex>
            </PopoverTrigger>
            {navItem.children && (
              <PopoverContent boxShadow={'xl'} bg={`white`} rounded={'xl'} mt={4}>
                <Flex direction={'column'}>
                  {navItem.children.map((child, index) => (
                    <DesktopSubNav key={child.id} {...child} />
                  ))}
                </Flex>
              </PopoverContent>
            )}
          </Popover>
        </Flex>
      ))}
    </Flex>
  );
};
