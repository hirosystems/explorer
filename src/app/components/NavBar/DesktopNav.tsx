import { Stack } from '@/ui/Stack';
import { Box } from '@/ui/Box';
import { Popover } from '@/ui/Popover';
import { PopoverTrigger } from '@/ui/PopoverTrigger';
import { PopoverContent } from '@/ui/PopoverContent';
import { FC } from 'react';
import { DesktopSubNav } from './DesktopSubNav';
import { NavItem } from './types';
import { Flex } from '@/ui/Flex';
import { useColorMode } from '@chakra-ui/react';
import { BsChevronDown } from 'react-icons/bs';
import { Icon } from '@/ui/Icon';

export const DesktopNav: FC<{ navItems: NavItem[] }> = ({ navItems }) => {
  const colorMode = useColorMode().colorMode;
  return (
    <Stack direction={'row'} spacing={4}>
      {navItems.map(navItem => (
        <Flex key={navItem.id} alignItems={'center'}>
          <Popover trigger={'hover'} placement={'bottom-start'} isLazy>
            <PopoverTrigger>
              <Flex alignItems={'center'} gap={'2px'}>
                <Box
                  as="a"
                  href={navItem.href ?? '#'}
                  color={'#fff'}
                  fontSize="14px"
                  fontWeight={500}
                  _hover={{
                    textDecoration: navItem.children ? 'none' : 'underline',
                  }}
                >
                  {navItem.label}
                </Box>
                {navItem.children && (
                  <Icon
                    as={BsChevronDown}
                    transition={'all .25s ease-in-out'}
                    transform={false ? 'rotate(180deg)' : ''}
                    size={3}
                    color={'#fff'}
                    position={'relative'}
                    top={'2px'}
                  />
                )}
              </Flex>
            </PopoverTrigger>
            {navItem.children && (
              <PopoverContent boxShadow={'xl'} bg={`bg.${colorMode}`} rounded={'xl'} mt={'16px'}>
                <Flex direction={'column'}>
                  {navItem.children.map((child, index) => (
                    <DesktopSubNav
                      key={child.id}
                      hasDivider={index < (navItem.children?.length || 0) - 1}
                      {...child}
                    />
                  ))}
                </Flex>
              </PopoverContent>
            )}
          </Popover>
        </Flex>
      ))}
    </Stack>
  );
};
