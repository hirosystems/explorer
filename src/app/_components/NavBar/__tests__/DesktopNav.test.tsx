import '@testing-library/jest-dom';
import { render } from '@testing-library/react';
import React from 'react';

import { DesktopNav } from '../DesktopNav';

describe('DesktopNav', () => {
  it('should render correctly', () => {
    const mockNavItems = [
      {
        id: '1',
        label: 'Sample Item 1',
        href: '#',
        children: [
          {
            id: '1.1',
            label: 'Sub Item 1',
            href: '#',
          },
        ],
      },
      {
        id: '2',
        label: 'Sample Item 2',
        href: '#',
      },
    ];

    const { asFragment } = render(<DesktopNav navItems={mockNavItems} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
