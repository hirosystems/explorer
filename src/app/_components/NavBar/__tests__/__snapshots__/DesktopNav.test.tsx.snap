// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DesktopNav should render correctly 1`] = `
<DocumentFragment>
  <div
    class="css-ffmerx"
  >
    <div
      class="css-70qvj9"
    >
      <div
        aria-controls="popover-content-:r1:"
        aria-expanded="false"
        aria-haspopup="dialog"
        class="css-j93dqc"
        id="popover-trigger-:r1:"
      >
        <a
          class="chakra-link css-uzv5li"
          href="#"
        >
          Sample Item 1
        </a>
        <svg
          class="chakra-icon css-g16cky"
          fill="currentColor"
          focusable="false"
          height="3.5"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 256 256"
          width="3.5"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M213.66,101.66l-80,80a8,8,0,0,1-11.32,0l-80-80A8,8,0,0,1,53.66,90.34L128,164.69l74.34-74.35a8,8,0,0,1,11.32,11.32Z"
          />
        </svg>
      </div>
      <div
        class="chakra-popover__popper css-0"
        style="visibility: hidden; position: absolute; min-width: max-content; inset: 0 auto auto 0;"
      >
        <section
          class="chakra-popover__content css-1kjb8d7"
          id="popover-content-:r1:"
          role="tooltip"
          style="transform-origin: var(--popper-transform-origin); opacity: 0; visibility: hidden; transform: scale(0.95) translateZ(0);"
          tabindex="-1"
        />
      </div>
    </div>
    <div
      class="css-70qvj9"
    >
      <div
        aria-controls="popover-content-:r3:"
        aria-expanded="false"
        aria-haspopup="dialog"
        class="css-j93dqc"
        id="popover-trigger-:r3:"
      >
        <a
          class="chakra-link css-yne82k"
          href="#"
        >
          Sample Item 2
        </a>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
